spec/classes/agent_spec.rb

```diff
@@ -1,281 +1,318 @@
-      function puppet_operational_dashboards::hosts_with_profile($profile) { return ['localhost.foo.com'] }
-      class{ 'puppet_operational_dashboards':
-        influxdb_host => 'localhost.foo.com',
-        include_pe_metrics => true,
-      }
-    PRE_COND
-  end
-
   on_supported_os.each do |os, os_facts|
     context "on #{os}" do
       let(:facts) { os_facts }
 
-      it { is_expected.to compile }
-    end
-  end
-
-  context 'when using default parameters' do
-    let(:influxdb_v2) do
-      {
-        'influxdb_v2' => [
+      context 'when using default parameters' do
+        let(:pre_condition) do
+          <<-PRE_COND
+            function puppet_operational_dashboards::hosts_with_profile($profile) { return ['localhost.foo.com'] }
+            class{ 'puppet_operational_dashboards':
+              influxdb_host => 'localhost.foo.com',
+              include_pe_metrics => true,
+            }
+          PRE_COND
+        end
+        let(:influxdb_v2) do
           {
-            'tls_ca'               => '/etc/telegraf/ca.pem',
-            'tls_cert'             => '/etc/telegraf/cert.pem',
-            'insecure_skip_verify' => true,
-            'bucket'               => 'puppet_data',
-            'organization'         => 'puppetlabs',
-            'token'                => '$INFLUX_TOKEN',
-            'urls'                 => ['https://localhost.foo.com:8086']
-          },
-        ],
-      }
-    end
-
-    it {
-      is_expected.to contain_class('telegraf').with(
-        ensure: '1.25.3-1',
-        archive_location: 'https://dl.influxdata.com/telegraf/releases/telegraf-1.25.3_linux_amd64.tar.gz',
-        interval: '10m',
-        manage_service: false,
-        outputs: influxdb_v2,
-      )
-
-      is_expected.to contain_service('telegraf').with(ensure: 'running')
-      is_expected.to contain_service('telegraf').that_requires(['Class[telegraf::install]', 'Exec[puppet_telegraf_daemon_reload]'])
-
-      ['/etc/telegraf/puppet_ca.pem', '/etc/telegraf/puppet_cert.pem', '/etc/telegraf/puppet_key.pem', '/etc/telegraf/ca.pem', '/etc/telegraf/cert.pem', '/etc/telegraf/key.pem'].each do |cert_file|
-        is_expected.to contain_file(cert_file)
+            'influxdb_v2' => [
+              {
+                'tls_ca'               => '/etc/telegraf/ca.pem',
+                'tls_cert'             => '/etc/telegraf/cert.pem',
+                'insecure_skip_verify' => true,
+                'bucket'               => 'puppet_data',
+                'organization'         => 'puppetlabs',
+                'token'                => '$INFLUX_TOKEN',
+                'urls'                 => ['https://localhost.foo.com:8086']
+              },
+            ],
+          }
+        end
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_class('puppet_operational_dashboards::profile::dashboards') }
+        it { is_expected.to contain_grafana_dashboard('Filesync Performance') }
```

Comment:
Should these `grafana_dashboard` tests be included here?  This class doesn't include them, and they're already covered in the `dashboards_spec.rb` tests since that is the class that manages them.

Reply:
yeah. The resource coverage from that class wasn't great and I added a few more tests. The resources are in the catalog because `puppet_operational_dashboards` is in the precondition. if you like I can remove it from the tests.

Reply:
Please do, IMO they should only live in `dashboards_spec.rb` since the dashboards class is the one that manages them.